tactics/redexpr.ml

```diff
@@ -129,9 +129,14 @@ type red_expr_val =
   (constr, Evaluable.t, constr_pattern, strength * RedFlags.reds) red_expr_gen0
 
 let make_flag_constant = function
-  | Evaluable.EvalVarRef id -> fVAR id
-  | Evaluable.EvalConstRef sp -> fCONST sp
-  | Evaluable.EvalProjectionRef p -> fPROJ p
+  | Evaluable.EvalVarRef id -> [fVAR id]
+  | Evaluable.EvalConstRef sp ->
+      begin
+        match Structures.PrimitiveProjections.find_opt sp with
+        | None -> [fCONST sp]
+        | Some p -> [fCONST sp; fPROJ p]
+      end
+  | Evaluable.EvalProjectionRef p -> [fPROJ p; fCONST (Projection.Repr.constant p)]
```

Comment:
Not sure this is what we want, but @Janno and I can't think of an alternative. Basically, when you give delta flags, they somehow need to talk about both the projection and the compatibility constant. If it was just the projection, the compatibility constant is not unfolded (even though it should always be morally transparent).